### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by rshym**
- yaml: rework common_data
- xen-tools: move patch from domd to dom0
- Rename yaml
- yaml: update boot_artifacts

### CODE REVIEW

File: prod-devel-rcar-s4.yaml [hunk 1]
```diff
@@ -44,9 +44,6 @@ common_data:
 
   # Common configuration options for all yocto-based domains
   conf: &COMMON_CONF
-    - [SSTATE_DIR, "${TOPDIR}/../../../common_data/sstate"]
-    - [DL_DIR, "${TOPDIR}/../../../common_data/downloads"]
```
Comment (lorc) [inline prod-devel-rcar-s4.yaml:48]: I can understand why sstate can be buggy. But do we really need to separate downloads?

Comment (rshym) [reply prod-devel-rcar-s4.yaml:48]: You are right, I will update the commit.

Also, I will add more commits to this PR to address comments from the external mail.

So this PR is draft and [WIP] until testing ends.

**Comment by rshym** [comment]
Tested on the Whitehawk. Works as expected.

**Comment by lorc** [comment]
It is still in draft state,BTW.

**Comment by rshym** [comment]
Two more fixes are added (PCI passthrough and hardcode for domains' rotfs).
Successfully retested on the Whitehawk.

**Comment by lorc** [comment]
Merging without review, as this a bunch of hacks for demo branch

**Pull request merged**